lib/src/common/polymer_descriptor.dart

```diff
@@ -202,7 +203,11 @@ JsObject _jsType(Type type) {
       return context['String'];
     case 'Map':
     case 'JsObject':
-    default:
       return context['Object'];
+    default:
+      var test = new JsObject(context['Object']);
```

Comment:
You should just be able to do `return type;` here. Js won't know what to do with it, but I think that is ok, since its a dart type anyways.

A crazier idea, if the type mixes in JsProxy then you could potentially return the constructor for that proxy object here.